packages/edgeidentity/android/src/main/java/com/adobe/marketing/mobile/reactnative/edgeidentity/RCTAEPEdgeIdentityDataBridge.java

```diff
@@ -0,0 +1,132 @@
+import com.adobe.marketing.mobile.edge.identity.IdentityItem;
+
+import java.util.List;
+import java.util.Map;
+
+final class RCTAEPEdgeIdentityDataBridge {
+
+    //Identity Item
+    final private static String ID_KEY = "id";
+    final private static String IS_PRIMARY_KEY = "primary";
+    private static final String AEP_AUTH_STATE_KEY = "authenticatedState";
+
+    private static final String TAG = "RCTAEPEdgeIdentityDataBridge";
+
+    static WritableMap mapFromIdentityMap(final IdentityMap map) {
+        if (map == null) {
+            return null;
+        }
+
+        if (map.isEmpty()){
+            return new WritableNativeMap();
+        }
+
+        WritableMap identityMapAsWritableMap = new WritableNativeMap();
+
+            for (String namespace : map.getNamespaces()) {
+                List<IdentityItem> items = map.getIdentityItemsForNamespace(namespace);
+                WritableArray itemsAsArray = new WritableNativeArray();
+
+                for (IdentityItem item : items) {
+                    WritableMap itemAsWritableMap = new WritableNativeMap();
+
+                    itemAsWritableMap.putString(ID_KEY, item.getId());
+                    itemAsWritableMap.putString(AEP_AUTH_STATE_KEY, item.getAuthenticatedState().getName());
+                    itemAsWritableMap.putBoolean(IS_PRIMARY_KEY, item.isPrimary());
+
+                    itemsAsArray.pushMap(itemAsWritableMap);
+                }
+
+                if (itemsAsArray.size() != 0) {
+                    identityMapAsWritableMap.putArray(namespace, itemsAsArray);
+                }
+            }
+        return identityMapAsWritableMap;
+    }
+
+    static IdentityMap mapToIdentityMap(final ReadableMap map) {
+
+        if (map == null) {
+            return null;
+        }
+
+        IdentityMap identityMapFromReadableMap = new IdentityMap();
+        ReadableMap mapValue = map.getMap("items");
+
+        ReadableMapKeySetIterator iterator =  mapValue.keySetIterator();
+
+        //iterate namespaces list
+        while (iterator.hasNextKey()){
+            String namespace  = iterator.nextKey();
+            ReadableArray namespaceArray  = mapValue.getArray(namespace);
+
+            //iterate items
+            for (int i = 0; i < namespaceArray.size(); i++) {
+
+                ReadableMap itemsAsMap = namespaceArray.getMap(i);
+
+                IdentityItem item = mapToIdentityItem(itemsAsMap);
+                if (item != null)
+
+                identityMapFromReadableMap.addItem(item, namespace);
+            }
+        }
+
+        return identityMapFromReadableMap;
+    }
+
+    @SuppressLint("LongLogTag")
+    static IdentityItem mapToIdentityItem(ReadableMap map) {
+        if (map == null) {
+            return null;
+        }
+
+        try {
+            getNullableBoolean(map, IS_PRIMARY_KEY);
+        } catch (Exception e) {
+            Log.d(TAG, "primary requires to be a boolean type: " + e);
+            return null;
+        }
```

Comment:
This code seems not be required anymore since you call getNullableBoolean below

Reply:
When I set isPrimary =  "true", it runs into can't convert string to boolean error.  
I need to add this to silence that and provide error message.

Reply:
We should add an additional check inside getNullableBoolean, for example 
`data.hasKey(key) && data.getType(key) == ReadableType.Boolean)  ? data.getBoolean(key) : null;`

Reply:
Getting error "Attempt to invoke virtual method "boolean java.lang.Boolean.booleanValue() on a null object reference" when checking this inside of the getNullableBoolean.

Reply:
Ok, let's take a look at this and handle in the next PR you have.